commit 0ec9261447c52e79ab28e4c9ec594d2c6b040f3e (refs/changes/51/2751/3)
Author: Jerry Jelinek <jerry.jelinek@joyent.com>
Date:   2017-10-09T16:29:21+00:00 (2 years ago)
    
    OS-6388 lx: access(/proc/sys) reports as read-only
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/procfs/lx_prvnops.c b/usr/src/uts/common/brand/lx/procfs/lx_prvnops.c
index c94cbb9500..59fac45556 100644
--- a/usr/src/uts/common/brand/lx/procfs/lx_prvnops.c
+++ b/usr/src/uts/common/brand/lx/procfs/lx_prvnops.c
@@ -6038,9 +6038,16 @@ lxpr_doaccess(lxpr_node_t *lxpnp, boolean_t shallow, int mode, int flags,
 	int shift = 0;
 	proc_t *tp;
 
-	/* lx /proc is primarily a read only file system */
+	/*
+	 * lx /proc is primarily a read only file system
+	 * We handle LXPR_SYSDIR as a special case. At least 'systemd' expects
+	 * access() to report /proc/sys is writable, but we can't do that in
+	 * lxpr_is_writable since it breaks other code paths that check if they
+	 * can write there.
+	 */
 	if ((mode & VWRITE) && !lxpr_is_writable(type)) {
-		return (EROFS);
+		if (type != LXPR_SYSDIR)
+			return (EROFS);
 	}
 
 	if (type == LXPR_PIDDIR) {
